AudiusProject/audius-protocol
# Only delete disk on secondaries

### Description
Prevents deleting data from disk from nodes that were once a user's primary. This is to prevent any damage from the trackBlockchainId bug from being made permanent.



### Tests

1. Use `A seed` to create a user and upload a track
2. Reconfig the user off of their primary (ex: http://cn3_creator-node_1:4002/manually_update_replica_set?userId=2&newPrimarySpId=1&newSecondary1SpId=4&newSecondary2SpId=2)
3. Wait for orphaned data recovery to clean up the user's data, and make sure it fails the forceWipe sync with the reason 'abort_node_used_to_be_primary'
4. Perform the same reconfig for a non-primary, and it should successfully wipe the orphaned node

Note that we don't set self SP ID (spID env var), so the above doesn't technically work out of the box locally. I tested with a debugger to verify each step in both scenarios, though (primary and secondary).


### Monitoring - How will this change be monitored? Are there sufficient logs / alerts?
Monitor for syncs failing with `abort_node_used_to_be_primary`. Some should fail with this message, but other forceWipe=true syncs should pass (spot check some failures to make sure the orphaned node was a primary at some point according to the /users/history route).
